xl-v1.go

```diff
@@ -193,27 +199,66 @@ func (xl XL) DeleteVol(volume string) error {
 
 // ListVols - list volumes.
 func (xl XL) ListVols() (volsInfo []VolInfo, err error) {
-	// Pick the first node and list there always.
-	disk := xl.storageDisks[0]
-	volsInfo, err = disk.ListVols()
-	if err == nil {
-		return volsInfo, nil
+	quorumCount := 0
+	emptyCount := 0
+	for _, disk := range xl.storageDisks {
+		var vlsInfo []VolInfo
+		vlsInfo, err = disk.ListVols()
+		if err == nil {
+			if len(vlsInfo) == 0 {
+				emptyCount++
+			} else {
+				volsInfo = append(volsInfo, vlsInfo...)
```

Comment:
> Not sure its good idea to merge list volume output of all disks. instead we could have unique and identical (at least name) volsInfo of each disk and quorum has to meet to x.readQuorum.
> Merging is needed since due to inconsistent results from two different disks while valid, i am just merging them assuming that self-heal will heal them properly.

Keeping your recommendation for now, but we need to see. How to fix this.